webhooks/virtualmachine/validation/virtualmachine_validator.go

```diff
@@ -855,14 +857,36 @@ func (v validator) validateVolumeWithPVC(
 		claimName = vol.PersistentVolumeClaim.ClaimName
 	)
 
+	if vol.PersistentVolumeClaim.ReadOnly {
+		allErrs = append(allErrs, field.NotSupported(pvcPath.Child("readOnly"), true, []string{"false"}))
+	}
+
 	if claimName == "" {
-		allErrs = append(
-			allErrs,
-			field.Required(pvcPath.Child("claimName"), ""))
+		allErrs = append(allErrs, field.Required(pvcPath.Child("claimName"), ""))
+		return allErrs
 	}
 
-	if vol.PersistentVolumeClaim.ReadOnly {
-		allErrs = append(allErrs, field.NotSupported(pvcPath.Child("readOnly"), true, []string{"false"}))
+	if vol.PersistentVolumeClaim.InstanceVolumeClaim != nil {
+		return allErrs
+	}
+
+	// For now, make this check best effort. There's an ask to provide immediate feedback but
+	// ideally the validation webhooks should just deal with internal consistency.
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := v.client.Get(ctx, ctrlclient.ObjectKey{Name: claimName, Namespace: vm.Namespace}, pvc); err != nil {
+		return allErrs
```

Comment:
Ignore the not found error here in case the PVC is created after VM creation?

Reply:
Nvm, I guess we cannot really enforce this if we ignore the not-found error here.